src/services/rate/client/privatebank.rate.service.ts

```diff
@@ -0,0 +1,21 @@
+import {IRateService, PrivateBankRateData} from "../rate.service.interface";
+import axios from "axios";
+import logger from "../../../helpers/logger";
+import {PRIVATBANK_URL} from "../../../config/rate.api.const";
+
+export class PrivatebankRateService implements IRateService{
+
+    private readonly currency: string = 'USD';
+
+    async getExchangeRate(): Promise<number> {
+        const response = await axios.get<PrivateBankRateData[]>(PRIVATBANK_URL);
+        logger.info(`api.privatbank.ua - Response: ${JSON.stringify(response.data)}`);
+        const rateData = response.data.find(currency => currency.ccy === this.currency);
+        if (!rateData) {
+            throw new Error('Currency data not found');
```

Comment:
i saw this error a few times in you code so maybe you should create custom error with this message to avoid code duplication